autogen.sh: Use glibtoolize

On macOS, libtool is installed with a 'g' prefix to avoid a conflict
with macOS' own /usr/bin/libtool; as a consequence, libtoolize becomes
glibtoolize.

Upstream-Status: Inappropriate [configuration]
--- autogen.sh.orig	2023-02-11 11:56:43.000000000 +0100
+++ autogen.sh	2023-02-11 11:56:53.000000000 +0100
@@ -1,4 +1,4 @@
-libtoolize
+glibtoolize
 install -d m4
 aclocal -I m4
 autoheader
